MetadataExtractor/Formats/Jpeg/JpegDhtReader.cs

```diff
@@ -42,18 +43,23 @@ public sealed class JpegDhtReader : IJpegSegmentMetadataReader
     {
         ICollection<JpegSegmentType> IJpegSegmentMetadataReader.SegmentTypes => new[] { JpegSegmentType.Dht };
 
-        private HuffmanTablesDirectory directory = null;
-
         public DirectoryList ReadJpegSegments(IEnumerable<JpegSegment> segments)
         {
-
-            var directory = new HuffmanTablesDirectory();
-            foreach (var segment in segments)
+            // This Extract structure is a little different since we only want to return one HuffmanTablesDirectory
+            // for one-to-many segments
+            var directories = new List<Directory>();
+            if (segments.Count() > 0)
             {
-                Extract(new SequentialByteArrayReader(segment.Bytes), directory);
+                var directory = new HuffmanTablesDirectory();
+                foreach (var segment in segments)
+                {
+                    Extract(new SequentialByteArrayReader(segment.Bytes), directory);
+                }
+
+                directories.Add(directory);
             }
 
-            return new List<Directory> { directory };
+            return directories;
```

Comment:
It should be possible to avoid allocating a `List<>` in cases where no DNL segment exists.

For example:

```c#
class Directory
{
    internal static readonly DirectoryList EmptyList = new Directory[0];
}
```

Then here if `directory` is `null` just return `Directory.EmptyList`.

Note that in the general case we'd use `Array.Empty<Directory>()` but because we also target net35 that's not an option.

BTW this is a minor thing so don't worry if you don't think it's worth it.

Reply:
This does "work" but not cleanly. DirectoryList needs to also be defined in the Directory class but 'using' can't be inherited so all the other DirectoryList preprocessor conditionals still must exist.

I agree this is relatively minor so I'll have it return an empty list for now. If something changes in the future to make this easier, it can be revisited.

Reply:
`DirectoryList` is just a local alias, so you'd only need to copy it into the `Directory` class, not all its consumers too.